ethcore/light/src/client/header_chain.rs

```diff
@@ -359,11 +365,24 @@ impl HeaderChain {
 	/// will be returned.
 	pub fn block_header(&self, id: BlockId) -> Option<encoded::Header> {
 		let load_from_db = |hash: H256| {
-			match self.db.get(self.col, &hash) {
-				Ok(val) => val.map(|x| x.to_vec()).map(encoded::Header::new),
-				Err(e) => {
-					warn!(target: "chain", "Failed to read from database: {}", e);
-					None
+			let mut cache = self.cache.lock();
+
+			match cache.block_header(&hash) {
+				Some(header) => Some(header),
+				None => {
+					match self.db.get(self.col, &hash) {
+						Ok(dbValue) => {
+							dbValue.map(|x| x.to_vec()).map(encoded::Header::new)
```

Comment:
last thing: in Rust we usually use `snake_case` style. So `dbValue` should be `db_value`